### PR TITLE
Bw/bar graphs

### PR DESCRIPTION
**Pull request by Br3nda**
I think PR #688 is awesome, so here is it with the merge conflicts fixed.

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
> Color literals like #ffa500 should only be used in variable declarations; they should be referred to via variable everywhere else.

I don't think that's used anywhere else (its... orange); though we do have
{code}
$beige: #f3f1ee
$brown: #413f3b

$green:  #5f8e43
$blue:   #2f4365
$red:    #8e4d43
$orange: #b2685c
$yellow: #b2935c
{code}

Weirdly; #b2685c is completely not orange. I didn't quite see where it was used in the UI looking at the code only

**Comment by CloCkWeRX** [comment]
I  think with the d3 'similar code', its just going to be something we ignore. Specifically:
```
Similar code found in 1 other location (mass = 73)
NEW
    return d3.append('g')
      .attr("class", "bar-label")
      .selectAll("text")
      .data(bars.map(function(bar){ return bar.name;}))
      .enter()
```

```
To ignore this duplication issue modify your .codeclimate.yml file and add 16dbcb58d6caa7ccfe241417831ecfa6 to exclude_fingerprints. An example is shown below.

.codeclimate.yml
engines:
  # ... CONFIG CONTENT ...
  duplication:
    enabled: true
    # ... CONFIG CONTENT ...
    exclude_fingerprints:
    - 16dbcb58d6caa7ccfe241417831ecfa6
# ... CONFIG CONTENT ...
```

**Comment by CloCkWeRX** [comment]
```
for (i; i < data.bars.length; i++){
      barValues.push(data.bars[i].value)
    }
```
Kind of keen to see if we can ```barValues.push(data.bars.map(function(bar) { return bar.value; });```

File: spec/javascripts/support/jasmine.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+# src_files
```
Comment (CloCkWeRX) [inline spec/javascripts/support/jasmine.yml]: I wonder if spec/javascript will execute in travis automagically, or if we have to invoke something else.

Generally like jasmine for complex JS testing

Comment (pozorvlak) [inline spec/javascripts/support/jasmine.yml]: That would be *awesome*, but it's not the end of the world if we add another line to .travis.yml.

In general, I'm +10000 to us adding JavaScript unit tests via some means, and I'm not too bothered about which framework we use.

Comment (CloCkWeRX) [inline spec/javascripts/support/jasmine.yml]: Let's make #1125 also be 'make js tests work in travis'

**Comment by Br3nda** [comment]
Css isn't my thing. Anyone who can quickly work out how to appease code climate?

**Comment by pozorvlak** [comment]
The one-line duplication between `bar_group.js` and `width_scale.js` looks trivial, but actually it's one identical line embedded in two functions which are logically identical but just different enough to fool CodeClimate. So it would be good to factor that out. The function in question boils down to `bars.map(function(bar) { return bar.value; })`, which I see you're using in the *other* "similar code found" issue - can we rely on the `map` method existing?

The "similar code found between bar_group.js and bar_label_group.js" issue looks like a false positive to me - sure, we could factor out some common code, but the similarity is superficial so we wouldn't gain anything thereby. To ignore it, add `16dbcb58d6caa7ccfe241417831ecfa6` to `engines.duplication.exclude_fingerprints` in .codeclimate.yml, as follows:

```
engines:
  # ... CONFIG CONTENT ...
  duplication:
    enabled: true
    # ... CONFIG CONTENT ...
    exclude_fingerprints:
    - 16dbcb58d6caa7ccfe241417831ecfa6
# ... CONFIG CONTENT ...
```

File: spec/javascripts/support/jasmine.yml (hunk 1 above)
Comment (pozorvlak) [inline spec/javascripts/support/jasmine.yml]: It doesn't look like the tests are being run in Travis; if I run `rspec spec/javascripts` locally I get 0 tests executed. @Br3nda, how are you invoking the tests locally?

**Comment by pozorvlak** [comment]
Oh, @CloCkWeRX already said everything I did a fortnight ago. D'oh. Oh well, at least I'm somewhat up to speed with the issues now.

**Comment by CloCkWeRX** [comment]
> can we rely on the map method existing
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map - mostly.


Feature	Chrome	Firefox (Gecko)	Internet Explorer	Opera	Safari
Basic support	(Yes)	1.5 (1.8)	9	(Yes)	(Yes)

**Comment by pozorvlak** [comment]
We can use [jQuery.map](http://api.jquery.com/jquery.map/): we're already using jQuery.

**Comment by Br3nda** [comment]
Superceded

**Pull request closed without merging**